tempodb/encoding/vparquet3/wal_block.go

```diff
@@ -699,6 +699,7 @@ func (b *walBlock) FetchTagValues(ctx context.Context, req traceql.AutocompleteR
 		if err != nil {
 			return fmt.Errorf("error opening file %s: %w", page.path, err)
 		}
+		defer file.Close()
```

Comment:
Is the defer inside the loop enough, or do we want to close it out each iteration through `blockFlushes`?

Reply:
That's correct, it's inside the loop and all of them get closed at the end, which is not optimal.   This func and several others are all following the same pattern, and I was aiming to have minimum changes necessary.  But happy to make larger changes and defer as expected if we want.